pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,381 @@
+        self.lib = lib
+        self.child = child
+        self._as_parameter_ = ptr or self.lib.conduit_node_create(None)
+
+    def __del__(self):
+        if not self.child:
+            self.lib.conduit_node_destroy(self)
+
+    def __setitem__(self, key, value):
+        key = key.encode()
+        match value:
+            case str():
+                self.lib.conduit_node_set_path_char8_str(self, key,
+                                                         value.encode())
+            case ConduitNode():
+                self.lib.conduit_node_set_path_node(self, key, value)
+            case int():
+                self.lib.conduit_node_set_path_int64(self, key, value)
+            case float():
+                self.lib.conduit_node_set_path_float64(self, key, value)
+            case (np.ndarray() | np.generic()):
+                value = np.ascontiguousarray(value)
+                fn = getattr(self.lib,
+                             f'conduit_node_set_path_{value.dtype}_ptr')
+                fn(self, key, value.ctypes.data, value.size)
+            case list():
+                value = np.array(value, dtype=float)
+                self.lib.conduit_node_set_path_float64_ptr(self, key,
+                                                           value.ctypes.data,
+                                                           value.size)
+            case _:
+                ValueError('ConduitNode: __setitem__ type not supported')
+
+    def __str__(self):
+        self.lib.conduit_node_print(self)
+        return '\n'
+
+    def append(self):
+        ptr = self.lib.conduit_node_append(self)
+        return ConduitNode(self.lib, ptr, child=True)
+
+
+class AscentWrappers(LibWrapper):
+    _libname = 'ascent_mpi'
+
+    # Functions
+    _functions = [
+        (None, 'ascent_close', c_void_p),
+        (c_void_p, 'ascent_create', c_void_p),
+        (None, 'ascent_execute', c_void_p, c_void_p),
+        (None, 'ascent_open', c_void_p, c_void_p),
+        (None, 'ascent_publish', c_void_p, c_void_p),
+    ]
+
+
+class AscentPlugin(RegionMixin, BasePlugin):
+    name = 'ascent'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    # Element name mapping for conduit
+    bp_emap = {'hex': 'hex', 'pri': 'wedge', 'pyr': 'pyramid', 'quad': 'quad',
+               'tet': 'tet', 'tri': 'tri'}
+
+    # Ascent filters that implicitly require velocity expression
+    v_filter = ['qcriterion', 'vorticity']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+
+        # Get datatype
+        self.dtype = intg.system.backend.fpdtype
+
+        # Underlying system elements class
+        self.elementscls = intg.system.elementscls
+
+        # Set order for division
+        order = self.cfg.getint('solver', 'order')
+        divisor = defaultdict(lambda: self.cfg.getint(cfgsect,
```

Comment:
Pull out the division into its own variable to simplify the lambda.  Then can do `defaultdict(lambda: div, pyr=0)` on a single line.